pkg/R/sparkR.R

```diff
@@ -149,11 +149,15 @@ sparkR.init <- function(
     }
     f <- file(path, open='rb')
     sparkRBackendPort <- readInt(f)
+    monitorPort <- readInt(f)
     close(f)
     file.remove(path)
-    if (length(sparkRBackendPort) == 0) {
+    if (length(sparkRBackendPort) == 0 || sparkRBackendPort == 0 ||
+     length(monitorPort) == 0 || monitorPort == 0) {
       stop("JVM failed to launch")
     }
+    # never close this socket, JVM is waiting for it
+    .sparkREnv$monitorF <- socketConnection(port = monitorPort)
```

Comment:
Should we support start SparkContext and stop multiple times? If yes, we should not clear this variable during spakrR.stop(), or the connection will be GCed (close), the JVM will shutdown.

In Python, we could do that, especially in tests.